lib/chai/core/assertions.js

```diff
@@ -988,6 +988,12 @@ module.exports = function (chai, _) {
 
   function assertInstanceOf (constructor, msg) {
     if (msg) flag(this, 'message', msg);
+    var constructorType = constructor === null ? 'null' : typeof constructor;
```

Comment:
I've reimplemented part of `instanceof` algo to avoid getting `@@hasInstance`  twice.